src/Microsoft.IdentityModel.Logging/TextWriterEventListener.cs

```diff
@@ -83,7 +84,7 @@ public TextWriterEventListener(string filePath)
         public TextWriterEventListener(StreamWriter streamWriter)
         {
             if (streamWriter == null)
-                throw LogHelper.LogArgumentNullException("streamWriter");
+                throw LogHelper.LogExceptionMessage(new ArgumentNullException("streamWriter", String.Format(CultureInfo.InvariantCulture, LogMessages.IDX10000, "streamWriter"))); 
```

Comment:
While not architecturally pure, having LogHelper.LogArgumentNullException("...") create a new ArgumentNullException would reduce code change.